hyundai_kia_connect_api/KiaUvoApiCA.py

```diff
@@ -550,16 +551,19 @@ def check_last_action_status(self, token: Token, vehicle: Vehicle, action_id: st
         headers["accessToken"] = token.access_token
         headers["vehicleId"] = vehicle.id
         headers["transactionId"] = action_id
-        headers["pAuth"] = self._get_pin_token(token, vehicle)
+        headers["pAuth"] = token.last_action_pin_token
         response = requests.post(url, headers=headers)
         response = response.json()
 
-        last_action_completed = (
-            response["result"]["transaction"]["apiStatusCode"] != "null"
-        )
+        _LOGGER.debug(f"{DOMAIN} - Last action_status API response: {response}")
+
+        api_result = response["result"]["transaction"]["apiResult"]
+        last_action_completed = ( api_result != "P" )
         if last_action_completed:
-            action_status = response["result"]["transaction"]["apiStatusCode"]
-            _LOGGER.debug(f"{DOMAIN} - Last action_status: {action_status}")
+            if api_result == "C":
+                state = { "status": response['result']['vehicle'] }
+                _LOGGER.debug(f"{DOMAIN} - Last action completed, updating vehicle state with payload: {state}")
+                self._update_vehicle_properties_base(vehicle, state)
```

Comment:
Does the state vehicle response match a full cached call?  If not it will delete values from vehicle.

Reply:
Yes, the data (in my case) is identical to a full cached call. This is how the web app updates vehicle data after an action, presumably the mobile app works the same way

Reply:
Awesome!